classes/local/factor/object_factor_base.php

```diff
@@ -349,6 +349,10 @@ public function create_event_after_factor_setup($user) {
      * Override in child class if necessary.
      */
     public function post_pass_state() {
-        return;
+        // Update lastverified for factor.
```

Comment:
If you do this here, won't this mean that if you have say TOTP + Email, and you pass TOTP but fail email, this won't get updated?

I'm in two minds about this, strictly speaking that factor was verified even through over all you didn't pass and log in. In practice it's probably moot

Reply:
This is still WIP. The idea is input factors will execute a verified when they are verified, as this is a passive mode for no input factors, it defaults to forcing one when passing if the factor passed. Soon I will have a finalised PR and we can run through the workflow fully. Just battling some regressions atm

Reply:
So is this comment still valid?